guest/config/config.go

```diff
@@ -0,0 +1,58 @@
+/*
+   Copyright 2023 The Kubernetes Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+// Package config allows reading configuration from the WebAssembly host.
+package config
+
+import (
+	"unsafe"
+
+	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/internal/mem"
+)
+
+var (
+	readConfig func() string = func() string {
```

Comment:
ok I reworked this to add unit tests. This is the first time it is possible to test the logic around host imports.